From ba22e4a83efa6a9523fd10d127211e159d3b416f Mon Sep 17 00:00:00 2001
From: "Thakur,Santosh KrX" <santosh.krx.thakur@intel.com>
Date: Fri, 4 Sep 2015 16:44:47 +0530
Subject: [PATCH] SST: Intel: CHT: Fix for possible numeric overflow

It handles possible numeric overflow caused due to addition of two
32-bit data.

Change-Id: I4bc845c30c0e35893dee3fd421562b8a79840b4a
Tracked-on: https://jira01.devtools.intel.com/browse/GMINL-18154
Signed-off-by: Thakur,Santosh KrX <santosh.krx.thakur@intel.com>
---
 sound/soc/intel/sst/sst_app_interface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/intel/sst/sst_app_interface.c b/sound/soc/intel/sst/sst_app_interface.c
index 4ca935a..bc75bdf 100644
--- a/sound/soc/intel/sst/sst_app_interface.c
+++ b/sound/soc/intel/sst/sst_app_interface.c
@@ -106,7 +106,7 @@ static int sst_create_algo_ipc(struct snd_ppp_params *algo_params,
 					struct ipc_post **msg, int pvt_id)
 {
 	u32 header_size = 0;
-	u32 ipc_msg_size = sizeof(u32) + sizeof(*algo_params)
+	u64 ipc_msg_size = sizeof(u32) + sizeof(*algo_params)
 			 - sizeof(algo_params->params) + algo_params->size;
 	u32 offset = 0;
 
-- 
1.9.1

